common/src/main/java/com/wynntils/wynn/model/CompassModel.java

```diff
@@ -5,39 +5,73 @@
 package com.wynntils.wynn.model;
 
 import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.ClientTickEvent;
 import com.wynntils.mc.event.SetSpawnEvent;
 import com.wynntils.mc.objects.Location;
 import com.wynntils.mc.utils.McUtils;
 import com.wynntils.wynn.model.map.poi.MapLocation;
 import com.wynntils.wynn.model.map.poi.WaypointPoi;
 import java.util.Optional;
+import java.util.function.Supplier;
 import net.minecraft.core.BlockPos;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public final class CompassModel extends Model {
+    private static Supplier<Location> supplier = null;
     private static Location compassLocation = null;
 
     public static void init() {}
 
+    @SubscribeEvent
+    public void onTick(ClientTickEvent.Start e) {
+        if (supplier == null) return;
+
+        Location newLocation = supplier.get();
+
+        if (newLocation == null) { // drop location
+            reset();
+        } else if (compassLocation != newLocation) { // update location
+            compassLocation = newLocation;
+
+            if (McUtils.mc().level != null) {
```

Comment:
I am pretty sure client tick only happens if there is a level. But we can keep this.